src/cpp/src/tokenizer.cpp

```diff
@@ -195,11 +250,10 @@ class Tokenizer::TokenizerImpl {
         read_special_tokens_map(models_path);
         // Try to read tokenizer_config if some token ids or token str are not defined.
         read_tokenizer_config_if_necessary(models_path);
+        parse_if_exists(models_path / "tokenizer_config.json", m_chat_template);
+        parse_if_exists(models_path / "processor_config.json", m_chat_template);
+        parse_if_exists(models_path / "chat_template.json", m_chat_template);
```

Comment:
it's not easy to follow order of special tokens setting and chat_templates

maybe we can have 3 separate functions:
- create tokenizers
- setup_special_tokens
- setup_chat_template 

in this case, code will not look like spaghetti (IMO) and each action is done with a dedicated function, while now the logic / order is spread across several functions